app/uk/gov/hmrc/partnershipidentificationfrontend/controllers/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.partnershipidentificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.partnershipidentificationfrontend.config.AppConfig
+import uk.gov.hmrc.partnershipidentificationfrontend.service.{JourneyService, PartnershipService}
+import uk.gov.hmrc.partnershipidentificationfrontend.views.html.check_your_answers_page
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class CheckYourAnswersController @Inject()(mcc: MessagesControllerComponents,
+                                           view: check_your_answers_page,
+                                           val authConnector: AuthConnector,
+                                           journeyService: JourneyService,
+                                           partnershipService: PartnershipService)
+                                          (implicit val config: AppConfig, executionContext: ExecutionContext)
+  extends FrontendController(mcc) with AuthorisedFunctions {
+
+
+  def show(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised() {
+        val identifiers = for {
```

Comment:
Part of the simplification below
```suggestion
      for {
```